src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,796 @@
+
+                    conflicting_metrics_between_1_2 = []
+                    # check quality metrics params
+                    for metric, params_1 in qm_params_1.items():
+                        if params_1 != qm_params_2.get(metric):
+                            conflicting_metrics_between_1_2.append(metric)
+                    # check template metric params
+                    for metric, params_1 in tm_params_1.items():
+                        if params_1 != tm_params_2.get(metric):
+                            conflicting_metrics_between_1_2.append("template_metrics")
+
+                    conflicting_metrics += conflicting_metrics_between_1_2
+
+                    if len(conflicting_metrics_between_1_2) > 0:
+                        warning_message = f"Parameters used to calculate {conflicting_metrics_between_1_2} are different for sorting_analyzers #{analyzer_index_1} and #{analyzer_index_2}"
+                        if enforce_metric_params is True:
+                            raise Exception(warning_message)
+                        else:
+                            warnings.warn(warning_message)
+
+        unique_conflicting_metrics = set(conflicting_metrics)
+        return unique_conflicting_metrics
+
+    def load_and_preprocess_csv(self, paths):
+        self._load_data_files(paths)
+        self.process_test_data_for_classification()
+
+    def process_test_data_for_classification(self):
+        """
+        Processes the test data for classification.
+
+        Extracts the target variable and features from the loaded dataset.
+        It handles string labels by converting them to integer codes and reindexes the
+        feature matrix to match the specified metrics list. Infinite values in the features
+        are replaced with NaN, and any remaining NaN values are filled with zeros.
+
+        Raises
+        ------
+        ValueError
+            If the target column specified is not found in the loaded dataset.
+
+        Notes
+        -----
+        If the target column contains string labels, a warning is issued and the labels
+        are converted to integer codes. The mapping from string labels to integer codes
+        is stored in the `label_conversion` attribute.
+        """
+
+        # Convert string labels to integer codes to allow classification
+        new_y = self.y.astype("category").cat.codes
+        self.label_conversion = dict(zip(new_y, self.y))
+        self.y = new_y
+
+        # Extract features
+        try:
+            if (set(self.metric_names) - set(self.testing_metrics.columns) != set()) and self.verbose is True:
+                print(
+                    f"Dropped metrics (calculated but not included in metric_names): {set(self.testing_metrics.columns) - set(self.metric_names)}"
+                )
+            self.X = self.testing_metrics[self.metric_names]
+        except KeyError as e:
+            print("metrics_list contains invalid metric names")
```

Comment:
Not sure but this message will end up right at the top of the stack trace and may not be easily seen. If possible you could append it to `str(e)` something like `raise KeyError(f"{str(e)}, metrics list ...!")`